api-operator/pkg/controller/api/api_controller.go

```diff
@@ -480,36 +517,63 @@ func (r *ReconcileAPI) Reconcile(request reconcile.Request) (reconcile.Result, e
 			return reconcile.Result{}, errHpa
 		}
 
+		for t := 24; t > 0; t -= 1 {
+			time.Sleep(5 * time.Second)
+			getEndPointValue := mgw.ExternalIP(&r.client, instance, operatorMode, mgwSvc, controlIngressData, controlOpenshiftConf)
```

Comment:
we have to update the value of `mgwSvc` by calling `k8s.Get()`, and then call the function `mgw.ExternalIP()`

Reply:
fixed in dc9a0dc